LBH.AdultSocialCare.Api/V1/UseCase/Common/Concrete/GetCarePackageReclaimUseCase.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LBH.AdultSocialCare.Api.V1.AppConstants.Enums;
+using LBH.AdultSocialCare.Api.V1.Boundary.Common.Response;
+using LBH.AdultSocialCare.Api.V1.Factories;
+using LBH.AdultSocialCare.Api.V1.Gateways.Common.Interfaces;
+using LBH.AdultSocialCare.Api.V1.UseCase.Common.Interfaces;
+
+namespace LBH.AdultSocialCare.Api.V1.UseCase.Common.Concrete
+{
+    public class GetCarePackageReclaimUseCase : IGetCarePackageReclaimUseCase
+    {
+        private readonly ICarePackageReclaimGateway _carePackageReclaimGateway;
+
+        public GetCarePackageReclaimUseCase(ICarePackageReclaimGateway carePackageReclaimGateway)
+        {
+            _carePackageReclaimGateway = carePackageReclaimGateway;
+        }
+
+        public async Task<CarePackageReclaimResponse> GetCarePackageReclaim(Guid carePackageId, ReclaimType reclaimType)
```

Comment:
Why do you need _reclaimType_? You already have carePackageId which is Guid unique.

Reply:
Since we store FNC and Care charge in single table I have used single use case and gateway method for retrieving FNC and Care Charges and In the Controller side there are two endpoints calling same use case with passing _reclaimType_ as a parameter. I think one single Care Package can have FNC and Care Charge at the same time.

Reply:
Understood. I misread that carePackageId is carePackageDetailId. Just in case, we may need to go with carePackageDetailId but let's keep it as-is now.